iotedge-ci.yml

```diff
@@ -0,0 +1,241 @@
+  - task: PublishBuildArtifacts@1
+    displayName: Publish build logs
+    condition: always()
+    inputs:
+      PathtoPublish: $(Build.ArtifactStagingDirectory)
+      ArtifactName: ubuntu
+      Parallel: true
+      ParallelCount: 8
+
+################################################################################
+- job: windows
+################################################################################
+  displayName: Windows
+  pool:
+    vmImage: vs2017-win2016
+  steps:
+  - task: NuGetToolInstaller@0
+    displayName: Install NuGet tool
+    inputs:
+      versionSpec: 4.x
+  - task: UsePythonVersion@0
+    displayName: Install Python 2.7
+    inputs:
+      versionSpec: 2.7.x
+      addToPath: true
+      architecture: x64
+  - task: NodeTool@0
+    displayName: Downgrade Node.js to 8.x for remote modules
+    inputs:
+      versionSpec: 8.x
+  - task: DownloadBuildArtifacts@0
+    displayName: Get Node.js dynamic library for binding
+    inputs:
+      buildType: specific
+      project: one
+      pipeline: v1 Node.js Library
+      buildVersionToDownload: latestFromBranch
+      branchName: refs/heads/master
+      downloadType: single
+      artifactName: nodejs-win2012r2
+      downloadPath: $(System.DefaultWorkingDirectory)
+      parallelizationLimit: 8
+  - task: AzureKeyVault@1
+    displayName: Get end-to-end test secrets
+    inputs:
+      azureSubscription: $(az.subscription)
+      keyVaultName: $(az.kv)
+      secretsFilter: >-
+        test-iothub-connection-string,
+        test-iothub-eventhub-connection-string,
+        test-x509-cert-base64,
+        test-x509-key-base64,
+        test-x509-thumbprint
+  - script: |
+      call "C:\\Program Files (x86)\\Microsoft Visual Studio\\2017\\Enterprise\\Common7\\Tools\\VsDevCmd.bat" ^
+        -arch=x64 -host_arch=x64
+
+      set "NODE_INCLUDE=%cd%\\nodejs-win2012r2\\inc"
+      set "NODE_LIB=%cd%\\nodejs-win2012r2\\lib"
+      set "IOTHUB_CONNECTION_STRING=$(test-iothub-connection-string)"
+      set "IOTHUB_EVENTHUB_CONNECTION_STRING=$(test-iothub-eventhub-connection-string)"
+      set "IOTHUB_E2E_X509_CERT_BASE64=$(test-x509-cert-base64)"
+      set "IOTHUB_E2E_X509_PRIVATE_KEY_BASE64=$(test-x509-key-base64)"
+      set "IOTHUB_E2E_X509_THUMBPRINT=$(test-x509-thumbprint)"
+      set IOTHUB_PARTITION_COUNT=4
+
+      call v1\\tools\\build.cmd ^
+        --platform x64 ^
+        --run-unittests ^
+        --run-e2e-tests ^
+        --enable-nodejs-binding ^
+        --enable-java-binding ^
+        --enable-dotnet-binding ^
+        --enable-dotnet-core-binding ^
+        --enable-java-remote-modules ^
+        --enable-nodejs-remote-modules
+    displayName: Run the build
+  - task: CopyFiles@2
+    displayName: Collect build logs
+    condition: always()
+    inputs:
+      SourceFolder: v1/deps
+      Contents: "**/build/?(output.txt|error.txt)"
+      TargetFolder: $(Build.ArtifactStagingDirectory)
+  - task: PublishBuildArtifacts@1
```

Comment:
Would it ever be useful to publish more than logs? Perhaps publishing the built bits for some sort of longer running integration test on Azure VMs?

Reply:
Realistically, I don't think we'll be adding any more build/test automation than what's here.